src/model_theory/rings/language.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2022 Joseph Hua. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Hua
+-/
+import model_theory.terms_and_formulas
+
+/-!
+# The language of rings
+
+## Main Definitions
+* A `first_order.language.ring.L` defines the language of rings,
+  which consists of `0`, `1`, `-`, `+`, `*`.
+-/
+universes u v u'
+
+namespace first_order
+namespace language
+namespace ring
+
+/-- The constant symbols in `first_order.language.ring.L` -/
+inductive consts : Type*
+| zero : consts
+| one : consts
```

Comment:
Is there a reason you have this in `Type*` and not `Type`? Is there any point in having it be universe polymorphic?

Reply:
not really I guess?

Reply:
Tbh I still am convinced that everything should be polymorphic whenever possible, and I don't understand why mathlib has chosen the naturals and booleans to not be so.

Reply:
That said, the reason the language library itself is so polymorphic is because you may have languages that depend on more complicated types, such as the language of vector spaces, which has a symbol for every scalar in some ring, which I would like to be able to live in any universe.

Reply:
you mentioned before that you might want to have some kind of general definition of "has the symbol one" and "has the symbol mul", which would then make instances of `has_one` and `has_mul` on the terms of the language. Given your point about vector spaces this general version should be polymorphic right?

Reply:
That said, it may not hurt to have these things be polymorphic.

Reply:
ok I'll put it in `Type` then